packages/extension-ui/src/hooks/useRequestsPagination.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2019-2023 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+const getSafeIndex = (index: number, arrLength: number) => Math.min(index, Math.max(arrLength - 1, 0));
+
+const useRequestsPagination = <T>(requests: Array<T>) => {
+  const [index, setIndex] = useState(0);
+
+  const next = () => setIndex((requestIndex) => Math.min(requestIndex + 1, requests.length));
+  const previous = () => setIndex((requestIndex) => Math.max(requestIndex - 1, 0));
+
+  useEffect(() => {
+    setIndex((currIndex) => getSafeIndex(currIndex, requests.length));
+  }, [requests]);
```

Comment:
I'm assuming `requests` is stable since it's used as a dep, but to make it slightly less suspicious, I guess you could just:
```suggestion
  }, [requests.length]);
```